File: Leetcode/sbgo/Leetcode867-转置矩阵.go
package sbgo

func transpose(matrix [][]int) [][]int {
	m, n := len(matrix), len(matrix[0])
	ans := make([][]int, n)
	for i := range ans {
		ans[i] = make([]int, m)
	}
	for i, row := range matrix {
		for j, v := range row {
			ans[j][i] = v
		}
	}
	return ans
}

File: Leetcode/Rust/Leetcode867-转置矩阵.rs
impl Solution {
    pub fn transpose(matrix: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let (m,n) = (matrix.len(),matrix[0].len());
        let mut ans = vec![vec![0;m];n];
        for i in 0..m{
            for j in 0..n{
                ans[j][i] = matrix[i][j]
            }
        }
        ans
    }
}